src/protocol/kademlia/impl/peer_routing_table_impl.cpp

```diff
@@ -88,25 +88,24 @@ namespace libp2p::protocol::kademlia {
 
     bucket.truncate(count);
 
-    return bucket.toVector();
+    return bucket.peerIds();
   }
 
-  outcome::result<bool> PeerRoutingTableImpl::update(const peer::PeerId &pid) {
+  outcome::result<bool> PeerRoutingTableImpl::update(const peer::PeerId &pid,
+                                                     bool is_permanent) {
     NodeId nodeId(pid);
     size_t cpl = nodeId.commonPrefixLen(local_);
 
     auto bucketId = getBucketId(buckets_, cpl);
     auto &bucket = buckets_.at(bucketId);
 
     // Trying to find and move to front
-    auto it = std::remove(bucket.rbegin(), bucket.rend(), pid);
-    if (it != bucket.rend()) {
-      *it = pid;
+    if (bucket.contains(pid)) {
```

Comment:
I'm not sure if it is enough. I guess peer ids in backet must be ordered by time of add/update.

Reply:
Thanks @xDimon for pointing this out.
From the go-libp2p implementation 
`// If the logical bucket to which the peer belongs is full and it's not the last bucket, we try to replace an existing peer
// whose LastSuccessfulOutboundQuery is above the maximum allowed threshold in that bucket with the new peer.
// If no such peer exists in that bucket, we do NOT add the peer to the Routing Table and return error "ErrPeerRejectedNoCapacity".`
Shall I try to implement it in a similar fashion?

Reply:
It sounds reasonable. Don't stop, please! 👍